templates/test/ci/cluster-template-prow-dual-stack.yaml

```diff
@@ -252,3 +252,75 @@ spec:
       - mv /etc/resolv.conf /etc/resolv.conf.OLD && ln -s /run/systemd/resolve/resolv.conf
         /etc/resolv.conf
       - systemctl restart systemd-resolved
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachinePool
+metadata:
+  name: ${CLUSTER_NAME}-mp-0
+  namespace: default
+spec:
+  clusterName: ${CLUSTER_NAME}
+  replicas: ${WORKER_MACHINE_COUNT}
```

Comment:
Maybe I have asked this question before: Should we use 2 variables for MachineDeployment and MachinePool in one template? If WORKER_MACHINE_COUNT == 2, we will have 4 Nodes in total.
It is good to have a template supporting both kinds of Machine but I think we can have a vmss only cluster, right?

Reply:
Yes, that's a good point. The `WORKER_MACHINE_COUNT` really means how many replicas per MachineDeployment or per MachinePool. Because even if it's VMSS only, you could have two or more MachinePools, each with `WORKER_MACHINE_COUNT` nodes. When the e2e tests wait for nodes to be Ready, they list all Machines, MachineDeployments, and MachinePools in the cluster so this number isn't used in any context as "total number of nodes" in any way (except maybe in ci-entrypoint.sh, but we should fix that). 

Having one env var per node group is not scalable. What if we wanted to test scale with 10,20, 50 MachinePools in one test template? So I think maybe we should rename that var to something more explicit like `REPLICAS_PER_POOL`, wdyt? That would be out of scope for this specific PR but something we can do separately.

> we can have a vmss only cluster

the main reason I'm putting them together is to reduce costs since we run lots of tests. In the future I think we want to move more towards a model where a single cluster is used to test several "node" features by using different node groups instead of separate clusters.

Reply:
> Having one env var per node group is not scalable. What if we wanted to test scale with 10,20, 50 MachinePools in one test template? So I think maybe we should rename that var to something more explicit like `REPLICAS_PER_POOL`, wdyt? That would be out of scope for this specific PR but something we can do separately.
Yes, renaming is a good idea. `REPLICAS_PER_POOL` seems to refer to MachinePool and how about `REPLICAS_PER_NODE_GROUP`?

> the main reason I'm putting them together is to reduce costs since we run lots of tests. In the future I think we want to move more towards a model where a single cluster is used to test several "node" features by using different node groups instead of separate clusters.
Yes, that's a good idea to reduce testing time.